packages/services/src/kernel/serialize.ts

```diff
@@ -3,13 +3,156 @@
 
 import * as KernelMessage from './messages';
 
+/**
```

Comment:
Could we please use `namespace Private` like we do in other packages for the private functions and all can we use the TypeScript camelCase conventions that we use elsewhere?